### PR TITLE
[feature] Adjust the JTAG/SWD frequency via cmdline option

### PR DESCRIPTION
**Pull request by chenguokai**
Fix https://github.com/stlink-org/stlink/issues/893
* Add --freq argument for st-flash

### CODE REVIEW

**Comment by chenguokai** [comment]
I found one missing problem in the test. I am fixing.

File: include/stlink/tools/flash.h [hunk 1]
```diff
@@ -26,9 +26,10 @@ struct flash_opts
     uint32_t val;
     size_t flash_size;	/* --flash=n[k][m] */
     int opt;
+    int freq;
```
Comment (Nightwalker-87) [inline include/stlink/tools/flash.h:29]: I think an inline comment would be useful here  (equal to `stlink.h`).

File: src/tools/flash.c [hunk 2]
```diff
@@ -28,9 +28,9 @@ static void cleanup(int signum) {
 
 static void usage(void)
 {
-    puts("command line:   ./st-flash [--debug] [--reset] [--opt] [--serial <serial>] [--format <format>] [--flash=<fsize>] {read|write} <path> [addr] [size]");
-    puts("command line:   ./st-flash [--debug] [--serial <serial>] erase");
-    puts("command line:   ./st-flash [--debug] [--serial <serial>] reset");
+    puts("command line:   ./st-flash [--debug] [--reset] [--opt] [--serial <serial>] [--format <format>] [--flash=<fsize>] [--freq=<Hz>] {read|write} <path> [addr] [size]");
+    puts("command line:   ./st-flash [--debug] [--freq=<Hz>] [--serial <serial>] erase");
+    puts("command line:   ./st-flash [--debug] [--freq=<Hz>] [--serial <serial>] reset");
```
Comment (Nightwalker-87) [inline src/tools/flash.c:33]: To have the documentation part of the related issue addressed we should also consider an update in our tutorial as well. In my opinion the section "Useful tool options" would be a good place. I wanted to extend that section anyway... There we can also include a table for the available frequency options as well.

File: tests/flash.c [hunk 3]
```diff
@@ -93,6 +95,51 @@ static struct Test tests[] = {
...
+        { .cmd = FLASH_CMD_WRITE,
+          .serial = { 0 },
+          .filename = "test.bin",
+          .addr = 0x80000000,
+          .size = 0,
+          .reset = 1,
+          .log_level = DEBUG_LOG_LEVEL,
+          .freq = 5,
+          .format = FLASH_FORMAT_BINARY }
+    },
```
Comment (Nightwalker-87) [inline tests/flash.c:133]: Duplicate?

Comment (chenguokai) [reply tests/flash.c:133]: The one is in another syntax (--freq=5k), the former one is "--freq 5k"

File: include/stlink/tools/flash.h (hunk 1 above)
Comment (chenguokai) [reply include/stlink/tools/flash.h:29]: I will add a comment in the coming commit tomorrow

File: src/tools/flash.c (hunk 2 above)
Comment (chenguokai) [reply src/tools/flash.c:33]: Indeed, I did some searching, no proper positions were found. Your idea sounds quite good👏

File: tests/flash.c (hunk 3 above)
Comment (Nightwalker-87) [reply tests/flash.c:133]: Correct, I recognise the slight difference now.

File: src/tools/flash.c (hunk 2 above)
Comment (Nightwalker-87) [reply src/tools/flash.c:33]: Feel free to add such info there at the top. I can do some additional formatting later on, when I continue my work on this tutorial. I may set up a table listing all available cmdline-options which also states for which tool they are available. That would avoid duplicates and provide a clearer appearance.

File: include/stlink/tools/flash.h (hunk 1 above)
Comment (chenguokai) [reply include/stlink/tools/flash.h]: Added

File: src/tools/flash.c (hunk 2 above)
Comment (chenguokai) [reply src/tools/flash.c:33]: Added, also for `--opt`

Comment (Nightwalker-87) [reply src/tools/flash.c:33]: Very attentive -  Thank you! :+1:

File: include/stlink/tools/flash.h (hunk 1 above)
Comment (Nightwalker-87) [reply include/stlink/tools/flash.h]: Thanks!

**Comment by Nightwalker-87** [comment]
:smiley: :+1:

**Pull request merged**